llama_index/embeddings/huggingface.py

```diff
@@ -201,3 +212,65 @@ def _get_text_embeddings(self, texts: List[str]) -> List[List[float]]:
         """Get text embeddings."""
         texts = [self._format_text(text) for text in texts]
         return self._embed(texts)
+
+
+class HuggingFaceInferenceAPIEmbeddings(HuggingFaceInferenceAPI, Embeddings):
```

Comment:
This should extend `BaseEmbedding`, not `Embeddings`

Then, you'll have to implement the proper `BaseEmbedding` functions so that it can be properly used as an embeddings model in llama-index

Reply:
I think it can be used properly as an embeddings model thanks to LangChain embeddings being [first class in Llama Index](https://github.com/run-llama/llama_index/blob/8f4c98225b85a523d33a46296269804f81dc5ba4/llama_index/embeddings/utils.py#L18).  `resolve_embed_model` will [cast it as a `LangchainEmbedding`](https://github.com/run-llama/llama_index/blob/8f4c98225b85a523d33a46296269804f81dc5ba4/llama_index/embeddings/utils.py#L75)

I like the interface of `LangChain` better, and also if I subclassed `BaseEmbeddings` it would mean we'd have to reimplement a ton of `Field`s, `__init__`, and test cases, which I am against. I think it's better this way

Reply:
I think in the near term, we want to remove langchain from the core dependencies though 👀

Reply:
I don't think you have to reimplement any fields or inits, the class extension can just swap Embedding for BaseEmbedding no?

Reply:
An aside is I just moved the `Embeddings` import to be from `langchain.bridge`.

I think even in the future without LangChain as primary dependency, it'd fine, because LlamaIndex can still have `LangchainEmbedding` as an option in `resolve_embed_model`. So to use `HuggingFaceInferenceAPIEmbeddings`, you would just need to `pip install huggingface_hub[inference] langchain`.

---

One question with `BaseEmbedding`, is what is the difference between `_get_query_embedding` and `_get_text_embedding`? The provided docstrings aren't giving an explanation of `query` vs `text`, and I see they're both `str`.

Reply:
Okay, I did some research and put it here: https://github.com/run-llama/llama_index/pull/8188

Reply:
Okay, also made https://github.com/run-llama/llama_index/pull/8189, when rolling this out, I found there were one too many repeated helper functions for `_format_query` capability

Reply:
I don't know about normalized embeddings, but they intentionally don't support batch embeddings (debate about batching here: https://github.com/huggingface/huggingface_hub/pull/1746).

I just pushed a migration to `BaseEmbedding`, let me know what you think 👍